File: Compare Version Numbers.py
class Solution:
    def compareVersion(self, version1: str, version2: str) -> int:
        v1 = [int(i) for i in version1.split(".")]
        v2 = [int(i) for i in version2.split(".")]
        n = max(len(v1), len(v2))
        while len(v1) < n:
            v1.append(0)
        while len(v2) < n:
            v2.append(0)

        if v1 < v2:
            return -1
        elif v2 < v1:
            return 1
        return 0

# class Solution:
#     def compareVersion(self, version1: str, version2: str) -> int:
#         v1, v2 = [int(i) for i in version1.split(".")], [int(i) for i in version2.split(".")]
#
#         def compare(a, b):
#             na, nb = len(a), len(b)
#             n = max(na, nb)
#             while len(a) < n:
#                 a.append(0)
#
#             while len(b) < n:
#                 b.append(0)
#
#             for i in range(n):
#                 if a[i] < b[i]:
#                     return -1
#                 elif a[i] > b[i]:
#                     return 1
#             return 0
#
#         return compare(v1, v2)

File: Compare Version Numbers.go
func compareStr(v1, v2 []string, n int) int {
    for i := 0; i < n; i++ {
        v1i, _ := strconv.Atoi(v1[i])
        v2i, _ := strconv.Atoi(v2[i])
        if v1i == v2i {
            continue
        }

        if v1i < v2i {
            return -1
        } else if v2i < v1i {
            return 1
        }
    }

    return 0
}

func compareVersion(version1 string, version2 string) int {
    v1 := strings.Split(version1, ".")
    v2 := strings.Split(version2, ".")
    n := max(len(v1), len(v2))
    for len(v1) < n {
        v1 = append(v1, "0")
    }

    for len(v2) < n {
        v2 = append(v2, "0")
    }

    return compareStr(v1, v2, n)
}
